<launch>
  <arg name="image"/>
  <arg name="autostart" default="True" />
  <arg name="num_rows" default="6" />
  <arg name="num_cols" default="8" />
  <arg name="dim" default="1" />
  <arg name="flags" default="0" />

  <arg name="viewer" default="True" />
  <arg name="marker_size" default="-1"/>
  <arg name="marker_width" default="-1"/>
  <arg name="scaling" default="-1" />
  
  <node type="calibration_detector"
        pkg="image_cb_detector"
        name="chessboard_detector"
        args="_autostart:=$(arg autostart) _num_x:=$(arg num_cols) _num_y:=$(arg num_rows) _dim_x:=$(arg dim) _dim_y:=$(arg dim) _flags:=$(arg flags)"
        output="screen"
        required="true">
    <remap from="image_raw" to="$(arg image)" />
  </node>

  <node type="image_annotator"
        pkg="image_cb_detector"
        name="calibration_annotator"
        args="_marker_size:=$(arg marker_size) _marker_width:=$(arg marker_width) _scaling:=$(arg scaling)"
        output="screen">
    <remap from="image" to="calibration_image" />
  </node>

  <node if="$(arg viewer)" type="image_view" pkg="image_view" name="$(anon image_view)">
    <remap from="image" to="annotated" />
  </node>
</launch>
